src/compas/datastructures/mesh/operations/weld.py

```diff
@@ -79,6 +93,97 @@ def mesh_unweld_vertices(mesh, fkey, where=None):
 
     return face
 
+def weld_mesh(mesh, tolerance = '3f'):
+    """Weld vertices of a mesh within some tolerance distance.
+
+    Parameters
+    ----------
+    mesh : Mesh
+        A mesh.
+
+    tolerance: str
+        Tolerance distance for welding.
+
+    Returns
+    -------
+    vertices : list
+        The vertices of the new mesh.
+    faces : list
+        The faces of the new mesh.
+

```

Comment:
yes, you would indeed expect the following:
`welded_mesh = mesh_weld_vertices(mesh)`

Reply:
I originally did this to have an external control on the class instead of having it as an input.

Reply:
But in that case, it should take as input also vertices and faces, so that the call is orthogonal (orthogonality in the API sense of the word, not the geometrical one!).